anvill/src/Lifters/FunctionLifter.cpp

```diff
@@ -1537,20 +1538,18 @@ llvm::Function *FunctionLifter::AddFunctionToContext(
 
   // The function we just lifted may call other functions, so we need to go
   // find those and also use them to update the context.
-  for (const auto &block : *func) {
-    for (const auto &inst : block) {
-      llvm::Function *called_func = nullptr;
-      if (auto call = llvm::dyn_cast<llvm::CallInst>(&inst)) {
-        called_func = call->getCalledFunction();
-      } else if (auto invoke = llvm::dyn_cast<llvm::InvokeInst>(&inst)) {
-        called_func = invoke->getCalledFunction();
-      }
-      if (called_func) {
-        const auto called_func_name = called_func->getName().str();
-        auto called_func_addr = AddressOfNamedFunction(called_func_name);
-        if (called_func_addr) {
-          lifter_context.AddEntity(called_func, *called_func_addr);
-        }
+  for (auto &inst : llvm::instructions(*func)) {
```

Comment:
We're now handling two different types of instructions. InstVisitor? If overhead is a concern:
>Note that this class is specifically designed as a template to avoid virtual function call overhead. Defining and using an InstVisitor is just as efficient as having your own switch statement over the instruction opcode. 
[Source](https://llvm.org/doxygen/classllvm_1_1InstVisitor.html)

Reply:
Alternatively there's casting to a [llvm::CallBase](https://llvm.org/doxygen/classllvm_1_1CallBase.html)

Reply:
Ahahaha performance on such a small thing isn't a concern :-P If it was I wouldn't create `std::string` here and there or heck, use LLVM :-P

Great suggestion on `CallBase`, made the change :-D